sasl/external/src/test/java/org/wildfly/security/sasl/external/ExternalSaslClientTest.java

```diff
@@ -213,13 +214,19 @@ public void testServerChallengeEmptyAuthzId() throws Exception {
     /**
      * Test failing (as we only authenticate "admin") authn for unsupported data "test" from client.
      */
-    @Test(expected = SaslException.class)
+    @Test
     public void testWrongServerChallenge() throws Exception {
         final SaslClientFactory factory = obtainSaslClientFactory(ExternalSaslClientFactory.class);
-        final SaslClient saslClient = factory.createSaslClient(MECHANISMS_EXTERNAL_ONLY, ADMIN, "test", "localhost", setProps(),
-                null);
-        assertFalse(saslClient.isComplete());
-        saslClient.evaluateChallenge("test".getBytes(StandardCharsets.UTF_8));
+        try {
+            final SaslClient saslClient = factory.createSaslClient(MECHANISMS_EXTERNAL_ONLY, ADMIN, "test", "localhost", setProps(),
+                    null);
+            assertFalse(saslClient.isComplete());
+            Assert.assertThrows(SaslException.class,()->{
+                saslClient.evaluateChallenge("test".getBytes(StandardCharsets.UTF_8));
+            });
+        }catch(SaslException saslException){
+            fail("Failed to create SaslServer Instance");
```

Comment:
@keshav-redhat  s/"Failed to create SaslServer Instance"/"Failed to create SaslClient Instance"